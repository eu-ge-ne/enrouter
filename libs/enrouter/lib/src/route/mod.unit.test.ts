import { describe, test, expect } from "vitest";

import { buildRoutesWithViteManifest } from "./mod.js";

import type { BuildRoutesWithViteManifestParams } from "./mod.js";

describe("buildRoutesWithViteManifest", () => {
  test("from 0 modules", () => {
    const params: BuildRoutesWithViteManifestParams = {
      modules: {},
      manifest: {},
      mapAssetUrl: (x) => x,
      entryId: "",
    };

    expect(buildRoutesWithViteManifest(params)).toMatchSnapshot();
  });

  test("from 1 module", () => {
    const params: BuildRoutesWithViteManifestParams = {
      modules: {
        "src/_layout.tsx": {
          path: "_layout.tsx",
          load: async () => undefined,
        },
      },
      manifest: {
        "src/main.tsx": {
          file: "assets/main.js",
          css: ["assets/main.css"],
        },
        "src/_layout.tsx": {
          file: "assets/_layout.js",
          imports: ["log.js"],
        },
        "log.js": {
          file: "assets/log.js",
        },
      },
      mapAssetUrl: (x) => x,
      entryId: "src/main.tsx",
    };

    expect(buildRoutesWithViteManifest(params)).toMatchSnapshot();
  });

  test("from 2 parent-child modules", () => {
    const params: BuildRoutesWithViteManifestParams = {
      modules: {
        "src/_layout.tsx": {
          path: "_layout.tsx",
          load: async () => undefined,
        },
        "src/abc/_layout.tsx": {
          path: "abc/_layout.tsx",
          load: async () => undefined,
        },
      },
      manifest: {
        "src/main.tsx": {
          file: "assets/main.js",
          css: ["assets/main.css"],
        },
        "src/_layout.tsx": {
          file: "assets/_layout.js",
          imports: ["log.js"],
        },
        "src/abc/_layout.tsx": {
          file: "assets/abc-_layout.js",
          imports: ["log.js"],
        },
        "log.js": {
          file: "assets/log.js",
        },
      },
      mapAssetUrl: (x) => x,
      entryId: "src/main.tsx",
    };

    expect(buildRoutesWithViteManifest(params)).toMatchSnapshot();
  });
});
