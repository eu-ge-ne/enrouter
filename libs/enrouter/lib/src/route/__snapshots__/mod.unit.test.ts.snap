// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildRoutes > from 0 modules 1`] = `undefined`;

exports[`buildRoutes > from 1 module 1`] = `
{
  "elements": {},
  "loaded": false,
  "modules": [
    {
      "fileName": "_layout.tsx",
      "id": "src/_layout.tsx",
      "load": [Function],
    },
  ],
  "path": "/",
  "test": {
    "keys": [],
    "pattern": /\\^\\(\\?=\\$\\|\\\\/\\)/i,
  },
}
`;

exports[`buildRoutes > from 2 parent-child modules 1`] = `
{
  "elements": {},
  "loaded": false,
  "modules": [
    {
      "fileName": "_layout.tsx",
      "id": "src/_layout.tsx",
      "load": [Function],
    },
  ],
  "path": "/",
  "test": {
    "keys": [],
    "pattern": /\\^\\(\\?=\\$\\|\\\\/\\)/i,
  },
  "tree": [
    {
      "elements": {},
      "loaded": false,
      "modules": [
        {
          "fileName": "_layout.tsx",
          "id": "src/abc/_layout.tsx",
          "load": [Function],
        },
      ],
      "path": "/abc",
      "test": {
        "keys": [],
        "pattern": /\\^\\\\/abc\\(\\?=\\$\\|\\\\/\\)/i,
      },
    },
  ],
}
`;
